android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1225,21 +1198,10 @@ public void customOnNavigateUp()
 
   void adjustCompass(int offsetY)
   {
-    Context context = getApplicationContext();
     if (mMapFragment == null || !mMapFragment.isAdded())
       return;
 
-    int resultOffset = offsetY;
-    //If the compass is covered by navigation buttons, we move it beyond the visible screen
-    if (mNavAnimationController != null && mNavAnimationController.isConflictWithCompass(offsetY))
-    {
-      int halfHeight = (int) (UiUtils.dimen(context, R.dimen.compass_height) * 0.5f);
-      int margin = UiUtils.dimen(context, R.dimen.margin_compass_top)
-                   + UiUtils.dimen(context, R.dimen.nav_frame_padding);
-      resultOffset = -(offsetY + halfHeight + margin);
-    }
-
-    mMapFragment.setupCompass(resultOffset, true);
+    mMapFragment.setupCompass(offsetY, true);
```

Comment:
Won't it be covered anymore?

Reply:
It is still necessary to adjust the compass position to show the navigation toolbar